plugins/gui/src/GUIWindows.cpp

```diff
@@ -94,394 +94,411 @@ bool GUIWindows::CreateContext_GL(megamol::core::CoreInstance* instance) {
 
 bool GUIWindows::PreDraw(glm::vec2 framebuffer_size, glm::vec2 window_size, double instance_time) {
 
-    // Check for initialized imgui api
-    if (this->api == GUIImGuiAPI::NO_API) {
-        megamol::core::utility::log::Log::DefaultLog.WriteError(
-            "[GUI] Found no initialized ImGui implementation. First call CreateContext_...() once. [%s, %s, line %d]\n",
-            __FILE__, __FUNCTION__, __LINE__);
-        return false;
-    }
-    // Check for existing imgui context
-    if (this->context == nullptr) {
-        megamol::core::utility::log::Log::DefaultLog.WriteError(
-            "[GUI] Found no valid ImGui context. [%s, %s, line %d]\n", __FILE__, __FUNCTION__, __LINE__);
-        return false;
-    }
-    // Set ImGui context
-    ImGui::SetCurrentContext(this->context);
-    // Check for existing fonts if shared between multiple contexts
-    ImGuiIO& io = ImGui::GetIO();
-    if (io.Fonts->Fonts.Size < 1) {
-        /// XXX TODO Solution for now is to shutdown megamol completely
-        /// Because if 'main' (= first) created imgui context is destroyed, fonts can not be restored for other
-        /// contexts?!
-        megamol::core::utility::log::Log::DefaultLog.WriteError(
-            "[GUI] Found no valid fonts. Maybe the ImGui context the fonts were shared with is destroyed. "
-            "[%s, %s, line %d]\n",
-            __FILE__, __FUNCTION__, __LINE__);
-        this->triggerCoreInstanceShutdown();
-        this->state.shutdown_triggered = true;
-        return false;
-    }
+    this->state.gui_post_drawing_enabled = this->state.gui_pre_drawing_enabled;
+    if (!this->state.gui_pre_drawing_enabled) {
+        // Reset ImGui context in core instance
+        if (this->core_instance != nullptr) {
+            this->core_instance->SetCurrentImGuiContext(nullptr);
+        }
+    } else {
```

Comment:
I suggest to do an early exit if drawing the GUI is disabled via `gui_pre_drawing_enabled` and `gui_post_drawing_enabled`. This leads to overall healthier code indentation and helps readability.

```
if(!drawing_enabled) {
    // do non-drawing stuff
    return;
}

// draw GUI
```

Reply:
Thanks, see commit 4720be1